backend/app/usecase/match/contains_all.go

```diff
@@ -0,0 +1,25 @@
+package match
+
+import "strings"
+
+var _ Keyword = (*ContainsAll)(nil)
+
+// ContainsAll checks if an input contains all the elements of a list.
+type ContainsAll struct {
+}
+
+// IsMatch checks if the input contains all the words.
+func (c ContainsAll) IsMatch(words []string, input string) bool {
+	if input == "" {
+		return false
+	} else if len(words) == 0 {
+		return false
```

Comment:
Logically speaking, shouldn't this be `true`?
`input` has all the items(in this case, no items) given in the `words`.
Again, this case is handled by the main logic below.

Reply:
You are right! I had in mind the search package and to optimize there I was thinking to reject if they are empty. Let me fix this now